### ISSUE TITLE
is it possible to support 1mbit flash patcher?

### ISSUE THREAD

**Issue by DrChai** [metroid-maniac/gba-flash-patcher#2]
AFAIK, 1Mbit Flash Save Cart only works with 1Mbit flash saves. I'd appreciate it if this repo also supports 1mbit patcher.
Im familiar with C programming but have no clue what those hex code stands for.

**Comment by theshillito** [comment]
I have a 32MB with 1Mbit flash save cartridge with RTC chip and a solar sensor for Boktai. I patched Drill Dozer (256K SRAM/FRAM 32KiB) using this tool and could save successfully without issue (except in mGBA when it hangs). Worth just trying it anyway.

**Comment by DrChai** [comment]
@theshillito i remembered I had tried some EEPROM games, like [Zelda - The Minish Cap (gba_eeprom_64k)](https://raw.githubusercontent.com/mamedev/mame/master/hash/gba.xml#:~:text=The%20Legend%20of%20Zelda%20%2D%20The%20Minish%20Cap%20(USA)) with a 1Mbit flash card. they all failed to save. Only some games with the slot type gba_flash_512 can play and save.

**Comment by theshillito** [comment]
@DrChai for eeprom games, use an SRAM patcher like the one in GBATA and then use gba-flash-patcher afterwards. I don't think gba-flash-patcher handles eeprom saves properly.

I tested this method with Minish Cap and saving works, albeit a bit slow.

**Comment by DrChai** [comment]
@theshillito You’re right! If I use GBATA to patch the SRAM first and then run this for a flash patch, the EEPROM game will be saveable. I thought it could directly patch an EEPROM game, as the description states. I hope there is a library available that can directly covert eeprom to flash, It would be much more convenient than getting a PC and GBATA.

**Comment by theshillito** [comment]
The most recent release has the note "Fix a small but critical bug that silently broke eeprom 64k" which makes me think that the bug wasn't fully fixed. Sadly, this project seems abandoned for now, so the GBATA workaround will probably have to do for now until someone who knows what they're doing can patch this up.

**Comment by DrChai** [comment]
@theshillito I took a quick look at the commit of the latest release, those are just a few typo fixes of C. I suppose it was not working for eeprom before. But like u said we have a workaround.

**Comment by danielepic96** [comment]
> Hello? I came across this tool and it almost seemed like the lifesaver I need, but unfortunately it doesn't work for the specific games I wanted (Battle Network series). Since they are already SRAM, I imagined I could just patch them immediately and they would work fine, but alas that is not the case, but I have no idea what to do about it. Any help on how I could make it work?

Hi, I recently made a fork. Try it. I had a problem with Kingdom Hearts Chain of Memories, now it’s working.

https://github.com/danielepic96/gba-flash-patcher
